pandas-stubs/core/frame.pyi

```diff
@@ -1357,7 +1357,7 @@ class DataFrame(NDFrame, OpsMixin):
     ) -> DataFrame: ...
     def astype(
         self,
-        dtype: _str | Dtype | dict[_str, _str | Dtype] | Series,
+        dtype: _str | Dtype | Mapping[_str, _str | Dtype] | Series,
```

Comment:
the keys can probably be any `Hashable(T)` as the columns might not be `str`.

Reply:
done in https://github.com/pandas-dev/pandas-stubs/pull/449/commits/5ff40dedffb393b03277c03492df9001e7fb1b23

Reply:
Could maybe add a test for a non-str column